repos/spack_repo/builtin/packages/py_immutables/package.py

```diff
@@ -26,6 +27,7 @@ class PyImmutables(PythonPackage):
 
     depends_on("python@3.5:", type=("build", "run"))
     depends_on("python@3.6:", type=("build", "run"), when="@0.16:")
+    depends_on("python@:3.12", type=("build", "run"), when="@:0.20")
```

Comment:
I think I see your concern, ~~which is that there is no 3.13 wheel~~. So we'd get something like
```suggestion
    # Based on PyPI wheel availability
    with default_args(type=("build", "link", "run")):
        depends_on("python@3.8:3.13", when="@0.21")
        depends_on("python@3.8:3.12", when="@0.20")
        depends_on("python@3.6:3.11", when="@0.19")
        depends_on("python@3.6:3.10", when="@0.17:0.18")
        depends_on("python@3.6:3.9", when="@0.16")
        depends_on("python@3.5:3.9", when="@0.15")
        depends_on("python@3.5:3.8", when="@0.14")
```
~~But 0.21 was released specifically for 3.13 support, see https://github.com/MagicStack/immutables/pull/118.~~

Reply:
0.21 has 3.13 wheels, but not 3.14 wheels. But yes, that is exactly the kind of structure I'm proposing.